onnxruntime/core/optimizer/bias_softmax_fusion.cc

```diff
@@ -59,9 +59,23 @@ bool TryBiasSoftmaxSubgraphMatch(Graph& graph, Node& start, Node*& add, Node*& s
     return false;
   }
 
+  // BiasSoftmax supports only float/float16/double - see ./onnxruntime/core/graph/contrib_ops/contrib_defs.cc
+  auto data_type1 = input1->TypeAsProto()->tensor_type().elem_type();
+  if (data_type1 != ONNX_NAMESPACE::TensorProto_DataType_DOUBLE &&
```

Comment:
nit:  given both the inputs use the same type constraint T so they will always have the same check - you can add a lambda or an inline function for this check and remove the duplicate check on line 70.